custom_components/google_home/sensor.py

```diff
@@ -132,17 +151,26 @@ def state(self) -> Optional[str]:
         return next_alarm.local_time_iso if next_alarm else STATE_OFF
 
     @property
-    def device_state_attributes(self):
+    def device_state_attributes(self) -> AlarmsAttributes:
         """Return the state attributes."""
         return {
             "alarms": self._get_alarms_data(),
             "integration": DOMAIN,
         }
 
-    def _get_alarms_data(self) -> List[Dict[Any, Any]]:
+    def _get_alarms_data(self) -> List[GoogleHomeAlarmDict]:
         """Update alarms data extracting it from coordinator"""
         device = self.get_device()
-        return self.as_dict(device.get_sorted_alarms()) if device else []
+        return (
+            [alarm.as_dict() for alarm in device.get_sorted_alarms()] if device else []
+        )
+
+
+class TimersAttributes(TypedDict):
+    """Typed dict for timers attributes"""
+
+    timers: List[GoogleHomeTimerDict]
+    integration: str
```

Comment:
I suggest we move out `TypedDict` to a separate file like `types.py`. It would give it a bit better structure